### DISCUSSION TITLE
Math in title/key

### DISCUSSION THREAD

**Discussion by ghbrown** [retorquere/zotero-better-bibtex#3037] [Q&A] [answered]
I'm struggling to get Zotero and BB to nicely handle an article title with math. The final bibtex entry is below
```
@article{lavrauwClassificationSubspaces$mathbbF^22017,
  title = {Classification of Subspaces in $\mathbb{F}^2 \otimes \mathbb{F}^3$
           and Orbits in $\mathbb{F}^2 \otimes \mathbb{F}^3 \otimes \mathbb{F}^r$},
  ... <irrelevant>
}
```
The `title` field is being handled fine (via `<script>`ing the two math sections), but special characters are appearing in the citation key.
The result is unchanged if I tag the item with `#LaTeX` and export.

I know there are probably *some* workarounds here:
- manually setting the export key
- changing the citation key formula (not working as of writing due to #3031)

but I'd like to get a recommendation for a preferred solution and also confirm that this is expected behavior. Thanks!

**Comment by retorquere** [comment]
#3031 is fixed, so please upgrade, then send a debug log from right-clicking that entry.

**Comment by ghbrown** [comment]
Updated, but BB still doesn't seem to be respecting my choice of "zotero" as the citation formula when I actually export. Instead the default BB formula is still being used somehow.
Anyway, one thing at a time. Here is the debug log: 5FC3XHBW-refs-euc/6.7.248-7

**Reply by retorquere** [reply, marked as answer]
I hadn't thought about this before, so it is expected only in the sense that I can confirm that this is what the current implementation will reliably do. The most straightforward way to get rid of it would be to use `Title.replace(/<script>.*?<\/script>/g, '')`. I can consider making that the default. 

`#LaTeX` affects bibtex production, not key production.

Can you retype `zotero` (or make any other change and come back to `zotero`) on the latest version and see whether that fixes things?

**Reply by ghbrown** [reply]
Thanks for confirming and for the workaround. That seems like a reasonable default to be, but others may disagree.
The special characters don't cause issues on the LaTeX side, but they are kind of unsettling and also inconvenient to type or remember.

Regarding the key formula issue, I've tried many things including your suggestions above, but still seem to be stuck on the BB default (regardless of what I put in that field). Happy to help triage further.

**Reply by retorquere** [reply]
Then it is a bug and you'll need to open an issue. I have tools built around GH issues.
